packages/block-editor/src/components/block-list/insertion-point.js

```diff
@@ -1,74 +1,119 @@
-/**
- * External dependencies
- */
-import classnames from 'classnames';
-
 /**
  * WordPress dependencies
  */
-import { useState } from '@wordpress/element';
 import { useSelect } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+import { Popover } from '@wordpress/components';
 
 /**
  * Internal dependencies
  */
 import Inserter from '../inserter';
 
-export default function BlockInsertionPoint( { rootClientId, clientId } ) {
-	const [ isInserterFocused, setInserterFocused ] = useState( false );
+function Indicator( { clientId } ) {
 	const showInsertionPoint = useSelect( ( select ) => {
 		const {
 			getBlockIndex,
 			getBlockInsertionPoint,
 			isBlockInsertionPointVisible,
+			getBlockRootClientId,
 		} = select( 'core/block-editor' );
+		const rootClientId = getBlockRootClientId( clientId );
 		const blockIndex = getBlockIndex( clientId, rootClientId );
 		const insertionPoint = getBlockInsertionPoint();
 		return (
 			isBlockInsertionPointVisible() &&
 			insertionPoint.index === blockIndex &&
 			insertionPoint.rootClientId === rootClientId
 		);
-	}, [ clientId, rootClientId ] );
+	}, [ clientId ] );
 
-	function onFocus( event ) {
-		// Stop propagation of the focus event to avoid selecting the current
-		// block while inserting a new block, as it is not relevant to sibling
-		// insertion and conflicts with contextual toolbar placement.
-		event.stopPropagation();
-		setInserterFocused( true );
+	if ( ! showInsertionPoint ) {
+		return null;
 	}
 
-	function onBlur() {
-		setInserterFocused( false );
+	return <div className="block-editor-block-list__insertion-point-indicator" />;
+}
+
+export default function InsertionPoint( {
+	className,
+	isMultiSelecting,
+	selectedBlockClientId,
+	children,
+} ) {
+	const [ isInserterShown, setIsInserterShown ] = useState( false );
+	const [ isInserterForced, setIsInserterForced ] = useState( false );
+	const [ inserterElement, setInserterElement ] = useState( null );
+	const [ inserterClientId, setInserterClientId ] = useState( null );
+
+	function onMouseMove( event ) {
+		if ( event.target.className !== className ) {
```

Comment:
What is this logic meant to be doing? It's quite unclear to me in reading this. A few inline comments or named utility functions could help clarify here.

FWIW, it raises some curiosity how reliable `className` is here, whether that might hurt maintainability or interoperability (I know of at least a few Chrome extensions which apply classes to the live DOM, etc). Could be something where [`event.target.classList.contains`](https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/contains) could prove more durable.

Reply:
Yeah, if we need to take into account outside changes to the class name, then checking with `contains` is better. Even better would be to keep a list of all the reference check if it's included.